### PR TITLE
Implementation of FlatMapFirst operator

### PR DESCRIPTION
**Pull request by troupmar**
`FlatMapFirst` operator is a special case of `flatMap` operator.
 
Like `FlatMapLatest`, it only allows one inner publisher at a time. Unlike `FlatMapLatest`, it will not cancel an ongoing inner publisher.
Instead it ignores events from the source until the inner publisher is done. It creates another inner publisher only when the previous one is done.

Sources:
- https://danielt1263.medium.com/rxswifts-many-faces-of-flatmap-5764e6c2018c
- https://www.oreilly.com/library/view/mastering-reactive-javascript/9781786463388/298ee42c-d800-4228-87dd-a42d2ae6e820.xhtml
- https://github.com/ReactiveX/RxSwift/blob/1a1fa37b0d08e0f99ffa41f98f340e8bc60c35c4/RxSwift/Observables/Merge.swift

### CODE REVIEW

File: Sources/Operators/FlatMapFirst.swift
Comment (freak4pc) [inline Sources/Operators/FlatMapFirst.swift:35]: Interesting idea for an implementation! Also seems like a good test suite to cover it.
Can you remove the explicit self here?

Comment (troupmar) [reply Sources/Operators/FlatMapFirst.swift]: Sure, it should be gone...

File: Tests/FlatMapFirstTests.swift
Comment (freak4pc) [inline Tests/FlatMapFirstTests.swift:62]: Can we change the naming to not use underscores? 
I believe that's not really common practice in Swift. 

Could also be a bit denser. Since we are in flatMapFirstTests, there's no need for `test_flatMapFirst` prefix. You can just call it `testErrorSkipsFlatmap`, or however you'd like. (This applies to the other tests too)

**Review by freak4pc — COMMENTED** [review]
Sorry for taking so long! I'd like to merge this, just a quick note on the test naming and we can go ahead. @troupmar 

If you don't have the time let me know and I'll do my best to take care of it myself.

File: Tests/FlatMapFirstTests.swift
Comment (troupmar) [reply Tests/FlatMapFirstTests.swift]: I got it renamed!

Comment (freak4pc) [reply Tests/FlatMapFirstTests.swift]: seems likje a minor conflict to fix and then we can merge :) thank you.

Comment (troupmar) [reply Tests/FlatMapFirstTests.swift]: My bad, sorry. I resolved it.

**Review by freak4pc — APPROVED** [review]

**Comment by freak4pc** [comment]
Still conflicted @troupmar :)

**Comment by troupmar** [comment]
There were new conflicts introduced, I guess. But I resolved them as well! Still waiting for "1 workflow awaiting approval" ;).

**Comment by freak4pc** [comment]
Thank you!

**Comment by troupmar** [comment]
Thanks for the assist! :)

**Pull request merged**
